modules/javafx.graphics/src/main/java/com/sun/javafx/font/directwrite/DWFactory.java

```diff
@@ -116,11 +118,29 @@ private static void checkThread() {
         }
     }
 
+    private static void addGraphicsPipelineDisposeHook(Runnable runnable) {
+        try {
+            Class plc = Class.forName("com.sun.prism.GraphicsPipeline");
+            Method gpm = plc.getMethod("getPipeline", (Class[])null);
+            Object plo = gpm.invoke(null);
+            Method adhm = plc.getMethod("addDisposeHook", Runnable.class);
+            adhm.invoke(plo, runnable);
```

Comment:
You can ignore this comment and leave it as you coded it. I added it before I read your earlier comment as to why you did it. I still don't see why reflection was used, but for consistency, it seems fine to do what you did.

Reply:
OK, I checked the history, and I see that the use of reflection pre-dates the current build structure. Prior to the switch to gradle, we used to have many small projects, rather than the current build where the graphics module is compiled as a single unit. In those days we sometimes used reflection to avoid a circular dependency. This is likely the reason that reflection was used here, since I can't think of any other good reason for it.

So you can either leave it as you have coded it, or dispense with the reflection and call GraphicsPipeline directly. I'll leave it up to you.

Separately, I might file a cleanup bug to eliminate the unnecessary uses of reflection, but that isn't a high priority.

Reply:
@kevinrushforth 
> So you can either leave it as you have coded it, or dispense with the reflection and call GraphicsPipeline directly. I'll leave it up to you.

Okay, since it does seem like there is no reason to use reflection now, I'll just use `GraphicsPipeline.getPipeline().addDisposeHook(...)` directly.